.github/workflows/pip_install.sh

```diff
@@ -0,0 +1,15 @@
+#!/bin/bash
+
+# A script to pip install a package with dependencies from requirements{suffix}.txt
+# if requirements{suffix}.txt exists.
+# Generates the requirements{suffix}.txt file once install is complete
+
+# usage: install_with_requirements.sh suffix [remaining parameters to pip install]
+
+suffix=$1
+shift
+
+touch requirements-${suffix}.txt
+pip install -r requirements-${suffix}.txt "${@}"
```

Comment:
It may be nice to treat these as constraints (`-c`) instead of requirements (`-r`) and check that the requirements file given was exhaustive, as so:
`diff requirements.txt <(pip freeze --exclude-editable)`
This eliminates the class of problems where the requirements being given no longer match what the packages themselves dictate. E.g. In the rare instance where I install `some-package` which used to depend on `vulnerable-dependency` but now uses `good-dependency` (despite being nominally the same version) pip will install both if given a requirements file with `-r`

Reply:
I've added this check and verified that it works as expected.